crates/polars-plan/src/dsl/function_expr/array.rs

```diff
@@ -209,6 +224,39 @@ pub(super) fn length(s: &Column) -> PolarsResult<Column> {
     Ok(c)
 }
 
+pub(super) fn slice(s: &Column, offset: i64, length: usize) -> PolarsResult<Column> {
+    let ca = s.array()?;
+
+    let series = ca.apply_to_inner(&|inner: Series| {
+        let sliced = inner.array()?.slice(offset, length);
```

Comment:
This is not correct. This calls into the same routine as `Series.slice` we want a new routine that takes a slice per array. Same for the `head` and `tail` implementations.